### ISSUE TITLE
linter: support multiple output formats simultaneously

### ISSUE THREAD

**Issue by rtarpine-vanta** [oxc-project/oxc#21574]
I want to run oxlint with the default output format for human-readable consumption, but also with json output for machine processing (e.g. we collect metrics on the rules that are violated), without running oxlint twice.

FYI, with ESLint this is possible via the ESLint API, which lets you load any formatter (either built-in or provided via a plugin) with https://eslint.org/docs/latest/integrate/nodejs-api#-eslintloadformatternameorpath and use it to generate new output given eslint's previous json output.

I can imagine oxlint gaining a flag to load previously-written json output and re-format it, instead of actually linting.

**Comment by csvn** [comment]
The `--format=github` used by default in GitHub actions produces very terse output, without the file or line that triggered the error. Since it's possible that there's lint errors for unchanged files (e.g. when adding new lint rules), just relying on the GitHub annotation for the PR diff view is not very helpful.

<img width="1053" height="512" alt="Image" src="https://github.com/user-attachments/assets/9018e605-8eac-4173-91a1-b7ec390f2dce" />

I would love to be able to produce two outputs at once, to avoid needing to run `oxlint` twice when there's errors:

```yml
    - id: oxlint
      run: npx oxlint
      # Print detailed errors only if the oxlint step failed
    - if: failure() steps.oxlint.outcome == 'failure'
      run: npx oxlint --format=default
```

**Comment by Stanzilla** [comment]
Another concrete use case from a large monorepo (~220 packages, GitLab CI): each `oxlint --type-check` run needs to produce **both** the GitLab Code Quality artifact (`--format gitlab`) and human-readable job logs. With a single `--format`, we have to shell-redirect the gitlab JSON to a file, which leaves the CI log empty — and re-running oxlint just for readable output doubles the cost of exactly the runs where `--type-check` is most expensive.

Our current workaround is a wrapper that captures `--format json` and post-processes it: converting diagnostics to the Code Quality schema ourselves and re-rendering a human summary (losing oxc's native rendering in the process). One detail worth noting for any design here: `--format json` is currently the only format that carries `help` text and column spans — `gitlab` drops both — so the json format being a superset of the other formats' data is what makes post-processing viable at all.

Either proposed shape would solve this for us:

- multiple simultaneous formats, e.g. `--format gitlab --output-file gl-codequality.json` (combining with #22945) while stdout keeps the default formatter — this matches how `eslint-formatter-gitlab` behaved (report file written as a side effect, stdout stays human-readable);
- or the "re-format previously-written json output" idea from the issue description, provided the json format stays a data superset.

**Comment by FarisTF** [comment]
Ran into the github format half of this in my own CI, with the same workaround as csvn (a second `--format=default` run on failure).

I did not attempt multi-format output, but #25020 makes the github format readable on its own: the message text now starts with `file:line:col`, the way Ruff does it. For my case that removes the need for the second run. The json and GitLab use cases here still need real multi-format support.
